lib/vmdb/appliance.rb

```diff
@@ -146,24 +146,40 @@ def self.get_network
       retVal
     end
 
-    def self.init_diagnostics
-      log_dir = "/var/www/miq/vmdb/log"
-      gem_log = File.join(log_dir, "gem_list.txt")
-      rpm_log = File.join(log_dir, "package_list_rpm.txt")
+    def self.installed_rpms
+      File.open(File.join(log_dir, "package_list_rpm.txt"), "a") do |file|
+        puts file, "start: date time is: #{Time.now.utc}"
+        LinuxAdmin.Rpm.list_installed.sort do |name, version|
+          puts file, "#{name} #{version}"
+        end
+      end
+    end
+
+    def self.installed_gems
+      File.open(File.join(log_dir, "gem_list.txt"), "a") do |file|
+        puts file, "start: date time is: #{Time.now.utc}"
+        puts file, `gem list`
+      end
+    end
 
+    def self.log_dir
+      "/var/www/miq/vmdb/log"
```

Comment:
If you wrap this in Pathname.new, then later File.join calls can be written nicer: `log_dir.join("gem_list.txt")`